base/collection_n1ql.go

```diff
@@ -119,8 +121,57 @@ func (c *Collection) CreatePrimaryIndex(indexName string, options *N1qlIndexOpti
 	return CreatePrimaryIndex(c, indexName, options)
 }
 
-func (c *Collection) WaitForIndexOnline(indexName string) error {
-	return WaitForIndexOnline(c, indexName)
+// WaitForIndexOnline takes set of indexes and watches them till they're online.
+func (c *Collection) WaitForIndexesOnline(indexNames []string, failfast bool) error {
+	logCtx := context.TODO()
+	mgr := c.cluster.QueryIndexes()
+	maxNumAttempts := 180
+	if failfast {
+		maxNumAttempts = 1
+	}
+	retrySleeper := CreateMaxDoublingSleeperFunc(maxNumAttempts, 100, 5000)
+	retryCount := 0
+
+	logged := make(map[string]bool)
+
+	indexOption := gocb.GetAllQueryIndexesOptions{
+		ScopeName:      c.ScopeName(),
+		CollectionName: c.Name(),
+	}
+
+	for {
+		onlineIndexCount := 0
+		currIndexes, err := mgr.GetAllIndexes(c.BucketName(), &indexOption)
+		if err != nil {
+			return err
+		}
+
+		for i := 0; i < len(currIndexes); i++ {
+			if currIndexes[i].State == IndexStateOnline {
+				if logged[currIndexes[i].Name] == false {
+					InfofCtx(logCtx, KeyAll, "Index %s is online", MD(currIndexes[i].Name))
+					logged[currIndexes[i].Name] = true
+				}
+			}
+		}
+
+		for _, listVal := range indexNames {
+			if logged[listVal] == true {
+				onlineIndexCount++
+			}
+		}
+
+		if onlineIndexCount == len(indexNames) {
```

Comment:
if you rename `logged` to `onlineIndexes` you actually know onlineIndexCount with `len(onlineIndexes)`